### DISCUSSION TITLE
dove_tool Profile 8 to 5 conversion for Mac support

### DISCUSSION THREAD

**Discussion by VCK02** [quietvoid/dovi_tool#253] [Q&A]
When playing profile 8 DV on Mac it shows as HDR10. However, with profile 5 it plays as Dolby Vision. Can dove_tool add support to convert profile 8 to 5.

**Comment by quietvoid** [comment]
No, that requires transcoding.

**Reply by VCK02** [reply]
I have m4v file having dolby profile 8 (Dolby Vision, Version 1.0, dvhe.08.07, BL+RPU, HDR10 compatible / SMPTE ST 2094 App 4, Version 1, HDR10+ Profile A compatible). First I demux it using it mp4demuxer which gave me two files i.e. out_2.ec3 and out_1.h265. These two files I did remuxing using mp4muxer using this command in terminal : ./mp4muxer -o output.mp4 -i out_2.ec3 -i out_1.h265 --dv-profile 5 --dvh1flag 1. This gave me profile 5 file (Dolby Vision, Version 1.0, dvhe.05.07, BL+RPU / SMPTE ST
2094 App 4, Version 1, HDR10+ Profile A compatible) but video and audio are out of sync (SC attached). Noticeable change that I have noticed is change in frame rate from variable (profile 8 sc attached) to Fixed frame rate (profile 5 sc attached). Is there any solution to it or is this a correct way to do? If you need any more information let me know will share. 
<img width="460" alt="DoVI Profile 8 Frame Rate" src="https://github.com/quietvoid/dovi_tool/assets/144264665/94b90435-7cb6-4b18-8e01-07b632559eed">
<img width="487" alt="DoVI Profile 5 Frame Rate" src="https://github.com/quietvoid/dovi_tool/assets/144264665/ebc872df-0c6e-4ecf-8967-2832b17a7959">
 
<img width="937" alt="Screenshot 2023-09-16 at 10 40 05 PM" src="https://github.com/quietvoid/dovi_tool/assets/144264665/ed4f6cb0-3b76-4c09-b535-9c84dceca61e">

**Reply by sw5163** [reply]
First, update your system and try to play that file in QuickTime.
[tvOS 17 now supports Dolby Vision 8.1](https://www.apple.com/newsroom/2023/06/tvos-17-brings-facetime-and-video-conferencing-to-apple-tv-4k/)
If that fails, check whether your file can play as DV in the latest Infuse.
[Now supports converting P8.1->P5 on the fly.](https://community.firecore.com/t/dolby-vision-profile-7-8-support-ts-mkv-files/19713/705)
If not, and you want to dig more into this:
[Apple Dolby Vision | AVS Forum](https://www.avsforum.com/threads/apple-dolby-vision.3252865/)

Regarding the frame rate problem:
https://forum.makemkv.com/forum/viewtopic.php?p=138790#p138790
If you don't want to change the frame rate at bitstream level:
`mp4box -add p8.1.hvc1.mp4:dvp=5 -new p5.dvh1.mp4`

**Reply by VCK02** [reply]
Thank you. Will try and let you know which works well.

**Reply by Sergij-SApcPro** [reply]
Hi, I know that this discussion is almost 2 years old now, but I found the **fix for the frame rate, and the de-synced audio.** 

Simply add `--input-video-frame-rate` to the command after the input video. Like this: `./mp4muxer -o output.mp4 -i out_2.ec3 -i out_1.h265 --input-video-frame-rate <original_frame_rate_here> --dv-profile 5 --dvh1flag 1`, and it really works! 

**Thanks** @VCK02 I can now **play the remuxed video from DV 8, to DV 5, on an LG Phone!**

Also for anyone else if the tool **mp4demuxer** doesn't work for you, and doesn't output anything, your video has to be **mp4**, so you can convert it to **mp4** with **ffmpeg**.

**Reply by VCK02** [reply]
<img width="505" height="501" alt="Screenshot 2026-05-09 at 3 40 28 PM" src="https://github.com/user-attachments/assets/ca7a953c-ef62-4630-bf18-08745b8d1487" />

So the thing is Video frame rate is mentioned as 23.976. However, there is a Min. as well as max. frame rate also. See attached screenshot.
So what to specify Min, max. or Main i.e.23.976 or all three?

**Comment by VCK02** [comment]
You can do it using this but it comes with codec string dvhe1 ( I am looking for codec string dvh1 or hvc1 to be compatible with QuickTime Player)

Step 1:  Extract raw HEVC
./mp4demuxer --input-file final_dv.mp4 --output-folder demux

You should get: demux/track_1.h265

Step 2: Extract RPU
dovi_tool extract-rpu -i demux/track_1.h265 -o RPU.bin

Step 3: Create Profile 5 conversion config
nano p5.json

{
"mode": 2
}

•	CTRL+O
•	ENTER
•	CTRL+X

Step 4: Convert RPU to Profile 5
dovi_tool editor -i RPU.bin -j p5.json -o RPU_p5.bin

Step 5: Inject Profile 5 RPU
dovi_tool inject-rpu -i demux/track_1.h265 --rpu-in RPU_p5.bin -o p5.hevc

Step 6: Extract audio
ffmpeg -i final_dv.mp4 -map 0:a:0 -c copy audio.eac3 

Better use mp4demuxer to get the dolby Atmos audio:     ./mp4demuxer --input-file 1.mp4 --output-folder tmp

Step 7: Mux Profile 5 MP4
./mp4muxer --dv-profile 5 --input-file p5.hevc --input-file audio.eac3 --output-file dv_p5.mp4

Step 8: Verify
ffprobe -show_streams dv_p5.mp4

You should now see: dv_profile=5 , codec_tag_string=dvhe1

This will only works if:  **ffprobe -show_streams dv_p8.mkv/mp4** command shows presence of dolby profile something like below:
[SIDE_DATA]
side_data_type=DOVI configuration record
dv_version_major=1
dv_version_minor=0
dv_profile=8
dv_level=6
rpu_present_flag=1
el_present_flag=0
bl_present_flag=1
dv_bl_signal_compatibility_id=1
[/SIDE_DATA]

if doesn't show using mkv file then simply remux to mp4 and see using ffprobe command.
